stdlib/2and3/socket.pyi

```diff
@@ -579,7 +579,7 @@ class socket:
 # ----- functions -----
 def create_connection(address: Tuple[Optional[str], int],
                       timeout: float = ...,
-                      source_address: Tuple[str, int] = ...) -> socket: ...
+                      source_address: Tuple[Union[bytearray, bytes, str], int] = ...) -> socket: ...
```

Comment:
The `str` should be `Text` in order to support 2.7.

Also, could you make the same change to `getaddrinfo`'s `host` argument below? `create_connection` calls `getaddrinfo`, so the type should be the same.

I confirmed that getaddrinfo takes both Text and bytes in the C code: https://github.com/python/cpython/blob/2.7/Modules/socketmodule.c#L4178.